it/src/test/java/com/google/cloud/teleport/it/bigquery/DefaultBigQueryResourceManagerTest.java

```diff
@@ -0,0 +1,373 @@
+        DefaultBigQueryResourceManager.builder(TEST_ID, PROJECT_ID);
+    assertThat(tmBuilder.getDatasetId())
+        .matches(TEST_ID.replace('-', '_') + "_\\d{8}_\\d{6}_\\d{6}");
+    tmBuilder
+        .setProjectId("fake_project_id")
+        .setTestId("fake_test_id")
+        .setDatasetId(DATASET_ID)
+        .setCredentials(credentials)
+        .setBigQuery(bigQuery);
+
+    assertThat(tmBuilder.getCredentials()).isEqualTo(credentials);
+    assertThat(tmBuilder.getProjectId()).isEqualTo("fake_project_id");
+    assertThat(tmBuilder.getTestId()).isEqualTo("fake_test_id");
+    assertThat(tmBuilder.getDatasetId()).matches(DATASET_ID);
+    assertThat(tmBuilder.getBigQuery()).isEqualTo(bigQuery);
+
+    DefaultBigQueryResourceManager tm = tmBuilder.build();
+
+    assertThat(tm.getProjectId()).isEqualTo("fake_project_id");
+    assertThat(tm.getTestId()).isEqualTo("fake_test_id");
+    assertThat(tm.getDatasetId()).matches(DATASET_ID);
+  }
+
+  @Test
+  public void testGetProjectIdReturnsCorrectValue() {
+    assertThat(testManager.getProjectId()).isEqualTo(PROJECT_ID);
+  }
+
+  @Test
+  public void testGetDatasetIdReturnsCorrectValue() {
+    DefaultBigQueryResourceManager.Builder tmBuilder =
+        DefaultBigQueryResourceManager.builder(TEST_ID, PROJECT_ID);
+    tmBuilder.setDatasetId(DATASET_ID).setCredentials(credentials);
+    DefaultBigQueryResourceManager tm = tmBuilder.build();
+
+    assertThat(tm.getDatasetId()).isEqualTo(DATASET_ID);
+  }
+
+  @Test
+  public void testGetTestIdReturnsCorrectValue() {
+    assertThat(testManager.getTestId()).isEqualTo(TEST_ID);
+  }
+
+  @Test
+  public void testCreateDatasetShouldThrowErrorWhenDatasetCreateFails() {
+    when(bigQuery.create((DatasetInfo) any())).thenThrow(RuntimeException.class);
+
+    assertThrows(
+        BigQueryResourceManagerException.class, () -> testManager.createDataset(DATASET_ID));
+  }
+
+  @Test
+  public void testCreateDatasetShouldNotCreateDatasetWhenDatasetAlreadyExists() {
+    prepareCreateDataset();
+    testManager.createDataset(DATASET_ID);
+
+    assertThrows(IllegalStateException.class, () -> testManager.createDataset(DATASET_ID));
+  }
+
+  @Test
+  public void testCreateDatasetShouldWorkWhenBigQueryDoesNotThrowAnyError() {
+    prepareCreateDataset();
+    testManager.createDataset(DATASET_ID);
+  }
+
+  @Test
+  public void testCreateTableShouldThrowErrorWhenTableNameIsNotValid() {
+    assertThrows(IllegalArgumentException.class, () -> testManager.createTable("", schema));
+  }
+
+  @Test
+  public void testCreateTableShouldThrowErrorWhenSchemaIsNull() {
+    assertThrows(IllegalArgumentException.class, () -> testManager.createTable(TABLE_NAME, null));
+  }
+
+  @Test
+  public void testCreateTableShouldCreateDatasetWhenDatasetDoesNotExist() {
+    prepareCreateDataset();
+    prepareCreateTable();
+
+    testManager.createTable(TABLE_NAME, schema);
+
+    verify(bigQuery).create((DatasetInfo) any());
```

Comment:
This may throw an unchecked cast exception, no? I usually see `any(DatasetInfo.class)` but if no warnings shown this is just a nitpick.

Reply:
Also a verify to make sure it runs `bigQuery.create(tableInfo)` would be great here